giraffe/README.md

```diff
@@ -0,0 +1,341 @@
+
+- **xTicks**: _array[number, ...]. Optional._ An array of values representing tick marks on the x-axis. Actual data values and axis scaling may cause Plot to not render all of the given ticks, or Plot rendering may extend beyond all of the rendered ticks.
+
+- **yTicks**: _array[number, ...]. Optional._ An array of values representing tick marks on the y-axis. Actual data values and axis scaling may cause Plot to not render all of the given ticks, or Plot rendering may extend beyond all of the rendered ticks.
+
+- **tickFont**: _string. Optional._ The [_CSS font_](https://developer.mozilla.org/en-US/docs/Web/CSS/font) value for the styling of the tick labels and axis labels.
+
+- **tickFontColor**: _string. Optional._ The _CSS color value_ of the tick labels and axis labels.
+
+- **valueFormatters**: _object. Optional._ An object containing column keys and their corresponding functions that format data for that column type. Each function takes a data value as the first argument, and an optional second argument as an options object. Returns a formatted string for that data value. For example:
+
+  <pre>
+  const config = {
+    // ...
+  
+    valueFormatters: {
+      _time: (t) => new Date(t).toLocaleTimeString(),  // ECMAScript time to human-readable time stamp
+      _value: (num) => num.toFixed(2),                 // values fixed to 2 decimal places
+    },
+  
+    // ...
+  }
+  </pre>
+
+- **xAxisLabel**: _string. Optional._ Uses **tickFont**, **tickFontColor**. Name to display below the x-axis.
+
+- **yAxisLabel**: _string. Optional._ Uses **tickFont**, **tickFontColor**. Name to display to the left of the y-axis.
+
+### Data properties
+
+- **table**: _Object. **Required**._ A data object produced by [Giraffe's utility functions](#utility-functions). Houses the data and getters and setters for that data, for the `<Plot>`.
+
+  - the `table` property of the return value from the `fromFlux` utility function.
+  - the return value from the `fromRows` utility function.
+  - the return value from the `newTable` utility function.
+
+- **layers**: _array[Object, ...]. **Required**._ An array of [LayerConfig objects](#layerconfig). These objects are customizations specific to a type of `<Plot>`. Currently, Giraffe supports only one pre-defined `<Plot>` type with any number of "custom layers". Custom layers are not pre-defined in Giraffe, but created through a callback render function in the configuration.
+
+- **xScale**: _"linear" | "log". Optional._ Sets the scaling of the x-axis by selecting the correct scaling function to be used by Layers. Linear scaling means the same distance between ticks represent the same increase in value. Logarithmic (log) scaling means the same distance between ticks can represent an exponential increase in value, used as a way to display data with a very wide range of values in a compact space.
```

Comment:
This is a little confusing. Does the `log` scaling mean `exponential` increases in value on Giraffe or `logarithmic`?

Reply:
I rewrote this part to be clearer.